msu/hooks/ai/tactical/behaviors/ai_switchto_ranged.nut

```diff
@@ -0,0 +1,31 @@
+::mods_hookExactClass("ai/tactical/behaviors/ai_switchto_ranged", function(o) {
+	local onExecute = o.onExecute;
+	o.onExecute = function( _entity )
+	{
+		_entity.getItems().m.IsMSUHandledItemAction = true;
+
+		local itemsBefore = {};
+		for (local i = 0; i < ::Const.ItemSlot.COUNT; i++)
+		{
+			itemsBefore[i] = _entity.getItems().getAllItemsAtSlot(i);
+		}
```

Comment:
That should probably be a generic 'getAllItemsInEverySlot' function

Reply:
There already exists in vanilla `item_container` a function called `getAllItems` which returns an array of all the items at all slots. However, in this particular case I want the items to be pulled into a table where each `key` is a `slot` and the values of that key are the items in that slot.